src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,67 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INITIAL_CAPACITY = 10;
+    private Object[] keys = new Object[INITIAL_CAPACITY];
+    private Object[] values = new Object[INITIAL_CAPACITY];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            for (int i = 0; i < size; i++) {
+                if (keys[i] == null) {
+                    values[i] = value;
+                    return;
+                }
+            }
+            ensureCapacity();
+            keys[size] = null;
+            values[size] = value;
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (key.equals(keys[i])) {
+                    values[i] = value;
+                    return;
+                }
+            }
+            ensureCapacity();
+            keys[size] = key;
+            values[size] = value;
+        }
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (key == null) {
+                if (keys[i] == null) {
+                    return (V) values[i];
+                }
+            } else if (key.equals(keys[i])) {
+                return (V) values[i];
+            }
```

Comment:
```suggestion
            if ((keys[i] == null && key == null) || (key != null && key.equals(keys[i]))) {
                return (V) values[i];
            }
```
There is example of simplifying with if statement, which you should do inside put method.